src/components/RangeSelector.js

```diff
@@ -92,6 +92,68 @@ class RangeSelector extends React.Component {
     });
   }
 
+  _handleClickStartOnTrack (e) {
+    const clientX = e.touches ? e.touches[0].clientX : e.clientX;
+
+    const newPixels = clientX - ReactDOM.findDOMNode(this.refs.rangeSelector).getBoundingClientRect().left;
+
+    //click point is less than lower selector
+    if (newPixels < this.state.lowerPixels) {
+      this.setState({
+        dragging: 'Lower'
+      });
+    }
+
+    //click point is higher than upper selector
+    if (newPixels > this.state.upperPixels) {
+      this.setState({
+        dragging: 'Upper'
+      });
+    }
+
+    //click point is higher than lower selector && also lower than midway point between selectors
+    if (newPixels > this.state.lowerPixels && newPixels < (this.state.lowerPixels + (this.state.upperPixels - this.state.lowerPixels) / 2)) {
+      this.setState({
+        dragging: 'Lower'
+      });
+    }
+
+    //click point is lower than upper selector && also greater than midway point between selectors
+    if (newPixels < this.state.upperPixels && newPixels > (this.state.upperPixels - (this.state.upperPixels - this.state.lowerPixels) / 2)) {
+      this.setState({
+        dragging: 'Upper'
+      });
+    }
+  }
+
+  _handleClickOnTrack (e) {
```

Comment:
This method looks almost identical to `_handleDragging`. Is there any way we can extract some of the shared logic into helper methods?